fabfile/dev.py

```diff
@@ -457,13 +546,127 @@ def run_test_big_object():
+        env_old['arakoon_bin'] = env.get('ARAKOON_189_BIN', '/usr/bin/arakoon')
+        if old_plugins:
+            arakoon_env = env_old
+        else:
+            arakoon_env = env
+
+        alba.demo_kill(env = env_old)
+
+        alba.arakoon_start(tls = tls, env = arakoon_env)
+        alba.wait_for_master(tls = tls, env = arakoon_env)
+
+        if old_plugins:
+            cmd = [
+                env_old['alba_bin'],
+                'apply-license',
+                env_old['license_file'],
+                env_old['signature'],
+                '--config', arakoon_config_file
+            ]
+            cmd_line = ' '.join(cmd)
+            local(cmd_line)
+
+        alba.maintenance_start(tls = tls)
+        alba.proxy_start(tls = tls)
+        alba.nsm_host_register_default(tls = tls)
+
+
+        if old_asds:
+            asd_tls = 'False'
+            asd_env = env_old
+        else:
+            asd_tls = tls
+            asd_env = env
+
+        kind = default_kind
+        alba.start_osds(kind, N, False, tls = asd_tls, env = asd_env)
+        alba.claim_local_osds(N, abm_cfg = arakoon_config_file, tls = tls)
+        #
+        test(old_proxy, old_plugins, old_asds, tls)
+
+
+    results = []
+    flavours = range(8)
+
+    t0 = time.time()
+    for flavour in flavours:
+        old_proxy   = flavour & 4 == 4
+        old_plugins = flavour & 2 == 2
+        old_asds    = flavour & 1 == 1
+        t0_test = time.time()
+
+        result = False
+        try:
+            deploy_and_test(old_proxy,old_plugins,old_asds)
+            result = True
+        except:
+            pass
+
+        t1_test = time.time()
+        delta_test = t1_test - t0_test
+        test_name = "flavour_%i" % flavour
+        results.append((test_name, result, delta_test))
+
+    failures = filter(lambda x: not x[1], results)
+    t1 = time.time()
+    delta = t1 - t0
+    # TODO: is there a better way than manually generate this?
```

Comment:
There is already dump_junit_xml() in alba.py. It currently dumps some dummy data and should provide a good starting point for what you're doing here.